drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/warehouse_changes_job.rb

```diff
@@ -0,0 +1,155 @@
+
+      count = 0
+      each_record_we_are_interested_in do |record|
+        count += 1
+        records_needing_processing << record
+      end
+
+      Rails.logger.info "Considering #{count} records"
+    end
+
+    def fetch_clients
+      Rails.logger.info 'Fetching client records'
+
+      personal_ids = records_needing_processing.map { |r| r['clientId'] }
+
+      self.clients = Hmis::Hud::Client
+        .where(c_t[:PersonalID].in(personal_ids))
+        .where(c_t[:data_source_id].eq(data_source.id))
+    end
+
+    def fetch_mci_unique_ids
+      Rails.logger.info 'Fetching MCI unique IDS and making a lookup table'
+
+      e_t = HmisExternalApis::ExternalId.arel_table
+
+      # by client_id
+      self.external_ids = HmisExternalApis::ExternalId
+        .where(e_t[:source_type].eq('Hmis::Hud::Client'))
+        .where(e_t[:source_id].in(clients.map(&:id)))
+        .where(namespace: NAMESPACE)
+        .to_a
+        .map { |eid| [eid.source_id, eid] }
+        .to_h
+    end
+
+    def upsert_changes
+      Rails.logger.info 'Upserting discovered changes'
+
+      insert_count = 0
+      update_count = 0
+      no_change_count = 0
+
+      client_by_personal_id = clients
+        .map { |c| [c.personal_id, c] }
+        .to_h
+
+      records_needing_processing.each do |record|
+        client = client_by_personal_id[record['clientId']]
+        external_id = external_ids[client.id]
+
+        if external_id.blank?
+          insert_count += 1
+          HmisExternalApis::ExternalId.create!(
+            value: record['mciUniqId'],
+            source: client,
+            namespace: NAMESPACE,
+            remote_credential: data_warehouse_api.send(:creds),
+          )
+        elsif external_id.value != record['mciUniqId']
+          update_count += 1
+          external_id.update_attribute(:value, record['mciUniqId'])
+        else
+          no_change_count += 1
+        end
+      end
+
+      Rails.logger.info "Inserted #{insert_count} MCI unique IDs"
+      Rails.logger.info "Updated #{update_count} MCI unique IDs"
+      Rails.logger.info "Ignored #{no_change_count} MCI unique IDs"
+    end
+
+    def merge_clients_by_mci_unique_id
+      e_t = HmisExternalApis::ExternalId.arel_table
+
+      self.merge_sets = HmisExternalApis::ExternalId
+        .where(e_t[:source_type].eq('Hmis::Hud::Client'))
+        .where(namespace: NAMESPACE)
+        .group(:value)
+        .having('count(*) > 1')
+        .select('value, array_agg(source_id ORDER BY source_id) AS client_ids')
+
+      Rails.logger.info "Found #{merge_sets.length} duplicate MCI unique IDs"
+
+      Rails.logger.info 'Enqueuing dedup jobs for each one'
+
+      merge_sets.each do |set|
+        Hmis::MergeClientsJob.perform_later(client_ids: set.client_ids, actor_id: actor_id)
+      end
+    end
+
+    def data_source
+      # FIXME: Is this right?
+      @data_source ||= GrdaWarehouse::DataSource.authoritative.where(name: 'HMIS').first!
```

Comment:
Nope it should be what's here - you should be able to access this class method  https://github.com/greenriver/hmis-warehouse/blob/2b7b1fefe7d9f7522d9c80d580cc56a395583641/drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis.rb#L3-L6